.changeset/brown-garlics-lie.md

```diff
@@ -0,0 +1,26 @@
+---
+"@latticexyz/world": major
+---
+
+All `World` methods now revert if the `World` calls itself.
+The `World` should never need to externally call itself, since all internal table operations happen via library calls, and all root system operations happen via delegate call.
+
+It should not be possible to make the `World` call itself as an external actor.
+If it were possible to make the `World` call itself, it would be possible to write to internal tables that only the `World` should have access to.
+As this is a very important invariance, we decided to codify it in an explicit requirement check in every `World` method, rather than of just relying on making it impossible to trigger the `World` to call itself.
```

Comment:
```suggestion
As this is a very important invariance, we made it explicit in a requirement check in every `World` method, rather than just relying on making it impossible to trigger the `World` to call itself.
```